janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/UncheckedSolrException.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.solr;
+
+/**
+ * @author davidclement@laposte.net
+ */
+public class UncheckedSolrException extends RuntimeException {
+
+    private static final long serialVersionUID = -8688401420333013730L;
+
+    /**
+     * @param msg Exception message
+     */
+    public UncheckedSolrException(String msg) {
```

Comment:
Change to `public UncheckedSolrException(SolrServerException cause)`.

Reply:
I keep this one but I change _public UncheckedSolrException(Throwable cause)_ to _public UncheckedSolrException(SolrServerException cause)_

Reply:
I'm looking at [java.io.UncheckedIOException ](https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html) as a model here. Requiring SolrServerException in all constructors protects against the misuse of the class for something other than its intended purpose (e.g. to wrap a SolrServerException as a RuntimeException). It also provides an API guarantee for future clients that the cause of an UncheckedSolrException is always a SolrServerException.

Reply:
Ok, I just make the change